sn_networking/src/event/swarm.rs

```diff
@@ -154,12 +154,19 @@ impl SwarmDriver {
                                 .collect(),
                         };
 
-                        self.relay_manager.add_potential_candidates(
-                            &peer_id,
-                            &addrs,
-                            &info.protocols,
-                            &self.bad_nodes,
-                        );
+                        let has_relayed = addrs.iter().any(|multiaddr| {
+                            multiaddr.iter().any(|p| matches!(p, Protocol::P2pCircuit))
+                        });
```

Comment:
This variable is used at two places, but can we possibly move this check inside `add_potential_candidates` just to keep logic at one place? 😄